### DISCUSSION TITLE
Highs executable needed?

### DISCUSSION THREAD

**Discussion by Edudiro** [pypsa-meets-earth/pypsa-earth#905] [Q&A]
Dear fellows,

I am trying to solve a network with Highs as a solver but it is giving me an error relatated to an executable (see below)

My question is, do we need the Highs executable? I come from pypsa-eur in which `linopy` is used and (if I recall corectly) it directly calls the highs interface for python `highspy`. No executable needed. Is pypsa-earth (which uses `pypsa.linopf`) different? 

INFO:pypsa.linopf:Total preparation time: 4.31s
INFO:pypsa.linopf:Solve linear problem using Highs solver
WARNING:pypsa.linopt:The HiGHS solver can potentially solve towards variables that slightly deviate from Gurobi, cbc, glpk
INFO:pypsa.linopt:Options: "{'primal_feasibility_tolerance': 0.0001, 'dual_feasibility_tolerance': 1e-05, 'ipm_optimality_tolerance': 1e-06, 'presolve': 'on', 'run_crossover': True, 'parallel': 'off', 'threads': 4, 'solution_file': 'C:\\Users\\SE91723\\AppData\\Local\\Temp\\pypsa-solvecjdq23t1.sol', 'write_solution_to_file': True, 'write_solution_style': 1, 'log_to_console': True, 'options': 'highs-default'}". List of options: https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.set
INFO:pypsa.linopt:Solver command: "highs --model_file C:\Users\SE91723\AppData\Local\Temp\pypsa-problem-ugff30jm.lp --solver ipm --options_file highs_options.txt"
Traceback (most recent call last):
  File "C:\Users\SE91723\Modelos\pypsa\pypsa-earth\.snakemake\shadow\tmpmlulmf_r\.snakemake\scripts\tmpaxridwxy.solve_network.py", line 575, in <module>
    n = solve_network(
  File "C:\Users\SE91723\Modelos\pypsa\pypsa-earth\.snakemake\shadow\tmpmlulmf_r\.snakemake\scripts\tmpaxridwxy.solve_network.py", line 527, in solve_network
    network_lopf(
  File "C:\Users\SE91723\Anaconda3\envs\pypsa-earth\lib\site-packages\pypsa\linopf.py", line 1576, in network_lopf
    res = solve(
  File "C:\Users\SE91723\Anaconda3\envs\pypsa-earth\lib\site-packages\pypsa\linopt.py", line 759, in run_and_read_highs
    process = subprocess.Popen(
  File "C:\Users\SE91723\Anaconda3\envs\pypsa-earth\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\SE91723\Anaconda3\envs\pypsa-earth\lib\subprocess.py", line 1457, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] El sistema no puede encontrar el archivo especificado

Thank you for your time!

**Comment by davide-f** [comment]
Hello and thanks for posting!
Currently, pypsa-earth does not rely on the most up-to-date pypsa version, which better supports highs.
In particular, the reason is that pypsa now fully maintains the high interface using linopy, but the older interface pypsa-earth still relies on is discontinued.
As https://github.com/pypsa-meets-earth/pypsa-earth/pull/796 is merged, then the constraint on pypsa can be lifted.
If you are from academia, I'd recommend to use gurobi in the meantime, alternatively you could use another open-source solver or the default glpk
